docs/docs/apis/core/hd-wallet-api.md

```diff
@@ -0,0 +1,298 @@
+            gapCount = 0; // Reset gap counter
+        } else {
+            gapCount++;
+        }
+        index++;
+    }
+    
+    return usedAddresses;
+}
+```
+
+### Key Management
+
+Access and manage wallet keys securely:
+
+```java
+// Get root key pair (if available)
+Optional<HdKeyPair> rootKeyPair = wallet.getRootKeyPair();
+if (rootKeyPair.isPresent()) {
+    HdKeyPair keyPair = rootKeyPair.get();
+    System.out.println("Root key available");
+}
+
+// Get root private key bytes
+Optional<byte[]> rootPrivateKey = wallet.getRootPvtKey();
+if (rootPrivateKey.isPresent()) {
+    byte[] keyBytes = rootPrivateKey.get();
+    System.out.println("Root private key: " + keyBytes.length + " bytes");
+}
+
+// Get stake address for delegation
+String stakeAddress = wallet.getStakeAddress();
+System.out.println("Stake Address: " + stakeAddress);
+```
+
+### Transaction Signing
+
+Sign transactions with wallet UTXOs:
+
+```java
+// Sign transaction with specific UTXOs
+Set<WalletUtxo> walletUtxos = getWalletUtxos(); // Your UTXO collection logic
+Transaction signedTx = wallet.signTransaction(transaction, walletUtxos);
+
+// Sign with stake key
+Transaction stakeSignedTx = wallet.signWithStakeKey(transaction);
+
+// Combined signing
+Transaction fullySignedTx = wallet.signTransaction(
+    wallet.signWithStakeKey(transaction), 
+    walletUtxos
+);
+```
+
+## Best Practices
+
+### Secure Wallet Creation
+
+```java
+// Generate wallet securely
+Wallet secureWallet = Wallet.create(Networks.mainnet());
+String mnemonic = secureWallet.getMnemonic();
+
+// Store mnemonic securely (encrypted storage recommended)
+// Never store mnemonic in plain text
+storeSecurely(mnemonic);
+
+// Recreate wallet when needed
+Wallet restoredWallet = Wallet.createFromMnemonic(Networks.mainnet(), mnemonic);
+```
+
+### Address Management Strategy
+
+```java
+// Organize addresses by purpose
+public class WalletManager {
+    private final Wallet wallet;
+    
+    public WalletManager(Network network, String mnemonic) {
+        this.wallet = Wallet.createFromMnemonic(network, mnemonic);
+    }
+    
+    // Receiving addresses (account 0)
+    public Address getReceivingAddress(int index) {
+        return wallet.getBaseAddress(0, index);
+    }
+    
+    // Change addresses (account 1)
+    public Address getChangeAddress(int index) {
+        return wallet.getBaseAddress(1, index);
+    }
+    
+    // Business addresses (account 2)
+    public Address getBusinessAddress(int index) {
+        return wallet.getBaseAddress(2, index);
+    }
+}
+```
+
+## HD Wallet vs Account API Comparison
+
+| Feature | HD Wallet API | Account API |
+|---------|---------------|-------------|
+| **Key Management** | Hierarchical key derivation | Single key pair |
+| **Address Generation** | Multiple derived addresses | Fixed addresses |
+| **Privacy** | High (new address per transaction) | Limited |
+| **Backup** | Single mnemonic | Individual keys |
```

Comment:
In Account API, we also have mnemonic. May be we can remove `Backup` to avoid confusion.
Also, let's remove `Standards`, as Account API also follows CIP1852 for key derivation.